src/top_level_id.rs

```diff
@@ -4,3 +4,18 @@ use crate::TransactionId;
 pub trait TopLevelId<Id: TransactionId>: Eq + PartialEq + Ord + PartialOrd {
     fn id(&self) -> Id;
 }
+
+
+impl<Id: TransactionId + Ord> TopLevelId<Id> for Id {
```

Comment:
```suggestion
impl<Id: TransactionId + Ord + Clone> TopLevelId<Id> for Id {
```
need clone here to clone the `self`